Neuro/Pathfinding/PathfindingHandler.cs

```diff
@@ -272,7 +276,7 @@ private Vector2[] RetracePath(Node startNode, Node endNode)
         }
 
         Vector2[] waypoints = path.Select(p => p.worldPosition).ToArray();
-        Array.Reverse(waypoints);
+        new Span<Vector2>(waypoints).Reverse();
```

Comment:
Are there any benefits for using Span.Reverse instead of Array.Reverse?

Reply:
In short we avoid doing 5 bounds/null checks, we cut out an Unsafe.Add call, and we avoid internally call a second function. This probably only saves a microsecond, but its still a difference.

Span.Reverse is defined as follows:
```cs
        public static void Reverse<T>(this Span<T> span)
        {
            if (span.Length <= 1)
            {
                return;
            }

            ref T first = ref MemoryMarshal.GetReference(span);
            ref T last = ref Unsafe.Add(ref Unsafe.Add(ref first, span.Length), -1);
            do
            {
                T temp = first;
                first = last;
                last = temp;
                first = ref Unsafe.Add(ref first, 1);
                last = ref Unsafe.Add(ref last, -1);
            } while (Unsafe.IsAddressLessThan(ref first, ref last));
        }
```

Array.Reverse(T[]) is defined as follows:
```cs
        public static void Reverse<T>(T[] array)
        {
            if (array == null)
                ThrowHelper.ThrowArgumentNullException(ExceptionArgument.array);
            Reverse(array, 0, array.Length);
        }

        public static void Reverse<T>(T[] array, int index, int length)
        {
            if (array == null)
                ThrowHelper.ThrowArgumentNullException(ExceptionArgument.array);
            if (index < 0)
                ThrowHelper.ThrowIndexArgumentOutOfRange_NeedNonNegNumException();
            if (length < 0)
                ThrowHelper.ThrowLengthArgumentOutOfRange_ArgumentOutOfRange_NeedNonNegNum();
            if (array.Length - index < length)
                ThrowHelper.ThrowArgumentException(ExceptionResource.Argument_InvalidOffLen);

            if (length <= 1)
                return;

            ref T first = ref Unsafe.Add(ref MemoryMarshal.GetArrayDataReference(array), index);
            ref T last = ref Unsafe.Add(ref Unsafe.Add(ref first, length), -1);
            do
            {
                T temp = first;
                first = last;
                last = temp;
                first = ref Unsafe.Add(ref first, 1);
                last = ref Unsafe.Add(ref last, -1);
            } while (Unsafe.IsAddressLessThan(ref first, ref last));
        }
```

Reply:
Thank you for the explanation. 

Given how close these two methods are to each other, I wonder if it's really worth reducing the readability of the code in order to avoid 1 Unsafe.Add call and 5 null/bounds checks. (We aren't really coding for performance here)

For now, I see no problem with keeping it as Span.Reverse, but I hope that won't confuse other contributors.

(Heck, I didn't even know about this method 😆)

Reply:
> Heck, I didn't even know about this method

Yeah, that's fair. I ran into it when trying to optimize another project and my profiler noted ~3% of the time was spent reversing arrays, so I switched to Span.Reverse() and the code got even faster.

> but I hope that won't confuse other contributors

I hope not either. To me it seems pretty straightforward, but I can see how it can be confusing to someone assuming it is a pure operation.